shared_model/backend/protobuf/commands/proto_add_asset_quantity.hpp

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_PROTO_ADD_ASSET_QUANTITY_HPP
+#define IROHA_PROTO_ADD_ASSET_QUANTITY_HPP
+
+#include "commands.pb.h"
+#include "cryptography/stub_hash.hpp"
+#include "interfaces/commands/add_asset_quantity.hpp"
+#include "utils/lazy_initializer.hpp"
+#include "backend/protobuf/common_objects/amount.hpp"
+
+namespace shared_model {
+  namespace proto {
+    class AddAssetQuantity final : public interface::AddAssetQuantity {
+     private:
+      template <typename Value>
+      using Lazy = detail::LazyInitializer<Value>;
+
+     public:
+      explicit AddAssetQuantity(const iroha::protocol::Command &command)
+          : AddAssetQuantity(command.add_asset_quantity()) {
+        if (not command.has_add_asset_quantity()) {
+          // TODO 11/11/17 andrei create generic exception message
+          throw std::invalid_argument(
```

Comment:
I don't really sure it's a good idea, factory that returns optional is definitely better
for now it's possible to use boost_assert

Reply:
I feel better solution provides only data classes that don't care about internal state and making verification strategy in the builder.